### PR TITLE
Is880/fix hanging

### LINKED ISSUE #880: Hangs while showing spinner - The "just refresh the page" issue
After login, retrieving the services fail and the page freezes showing the loading spinner. The only way out is to refresh the page.
Identify the problem and fix it.

@sanderegg Sees it happening in other screens and almost every time in prod.

### PR DESCRIPTION
**Pull request by pcrespov**
## What do these changes do?

Fixes this:
bug - login webserver 2019-11-13 14:19:39

call to ``/v0/me`` fails since auth layer raises an unhandled exception (500) due to a connection closed by the db!

- FIX: authorization module retries call to db upon failure. If still fails, then it responds [503](https://httpstatuses.com/503). @odeimaiz PLEASE check reaction of client to this response!
- enhanced log info on request that produced error
- Adds **application name** in db connections
image
- General cleanup of engine disposal
- Adds watchdog in development mode to auto reboot storage and webserver when files changed (thanks to @sanderegg )

- Graceful services shutdown (fixes based on input from https://hynek.me/articles/docker-signals/ )
   - Modified boot and entrypoints of services in production
   - added new test
- Changes in system tests: deploy done as fixture in tests

**NOTE:** please check in front end the reaction to a failing call to ``v0/me`` !

## Related issue number

Fixes #880 

## How to test

 - Not really reproducible all the time...
 - login/out repetitively

## Checklist

- [x] I think the code is well written
- [x] Unit tests for the changes exist
- [x] Documentation reflects the changes
- [x] **Runs in the swarm**
- [x] If you design a new module, add your user to .github/CODEOWNERS

### CODE REVIEW

File: services/web/server/tests/unit/with_postgres/test_users.py [hunk 1]
```diff
@@ -273,3 +275,43 @@ def client(loop, aiohttp_client, app_cfg, postgres_service):
...
+@pytest.fixture
+def mock_failing_connection(mocker) -> MagicMock:
+    """
+        async with engine.acquire() as conn:
+            await conn.execute(query)  --> will raise OperationalError
+    """
+    # See http://initd.org/psycopg/docs/module.html
+    conn_execute = mocker.patch.object(SAConnection, "execute")
+    conn_execute.side_effect=OperationalError("MOCK: server closed the connection unexpectedly")
+    return conn_execute
```
Comment (sanderegg) [inline services/web/server/tests/unit/with_postgres/test_users.py]: cool!

File: services/sidecar/src/sidecar/utils.py [hunk 2]
```diff
@@ -94,7 +94,10 @@ def __init__(self):
 class DbSettings:
     def __init__(self):
         self._db_config = db_config()
-        self.db = create_engine(self._db_config.endpoint, client_encoding='utf8', pool_pre_ping=True)
+        self.db = create_engine(
```
[the remark this answers is not in the document]
Comment (pcrespov) [reply services/sidecar/src/sidecar/utils.py]: I did not add this option. Was already there. Just added the application name

File: services/storage/docker/boot.sh [hunk 3]
```diff
@@ -26,11 +26,14 @@ then
   echo "  PIP :"
   $SC_PIP list | sed 's/^/    /'
 
+  #------------
+  echo "  setting entrypoint to use watchmedo autorestart..."
+  entrypoint='watchmedo auto-restart --recursive --pattern="*.py" --'
```
Comment (mguidon) [inline services/storage/docker/boot.sh]: Does this allow to remote ssh with vscode, work on the files and have immediate restart?

Comment (pcrespov) [reply services/storage/docker/boot.sh]: yes. @sanderegg  found the tool and I am adding it everywhere in development mode

**Pull request merged**